src/site/content/en/blog/url-protocol-handler/index.md

```diff
@@ -0,0 +1,267 @@
+mailto:Jane.Doe@example.com
+news:comp.infosystems.www.servers.unix
+https://web.dev/
+```
+
+The term _Uniform Resource Locator_ (URL) refers to the subset of URIs that, in addition to
+identifying a resource, provide a means of locating the resource by describing its primary access
+mechanism (e.g., its network location).
+
+## Background on the `registerProtocolHandler()` method
+
+The secure-content-only `Navigator` method
+[`registerProtocolHandler()`](https://developer.mozilla.org/en-US/docs/Web/API/Navigator/registerProtocolHandler)
+lets sites register their ability to open or handle particular URL schemes. Therefore, sites need to
+call the method like so: `navigator.registerProtocolHandler(scheme, url)`. The two parameters are
+defined as follows:
+
+- `scheme`: A string containing the protocol the site wishes to handle.
+- `url`: A string containing the URL of the handler. This URL must include `%s`, as a placeholder
+  that will be replaced with the escaped URL to be handled.
+
+The scheme must either be one of the
+[safelisted schemes](https://html.spec.whatwg.org/multipage/system-state.html#safelisted-scheme)
+(for example, `mailto`, `bitcoin`, or `magnet`) or begin with `web+`, followed by at least one or
+more lowercase ASCII letters after the `web+` prefix, for instance, `web+coffee`.
+
+To make this clearer, here is a concrete example of the flow:
+
+1. The user visits a site at `https://coffeeshop.example.com/` that makes the following call:
+   `navigator.registerProtocolHandler('web+coffee', 'coffee?type=%s')`.
+1. At a later point, while visiting `https://randomsite.example.com/`, the user clicks on a link
+   such as `<a href="web+coffee:latte-macchiato">All about latte macchiato</a>`.
+1. This causes the browser to navigate to the following URL:
+   `https://coffeeshop.example.com/coffee?type=web%2Bcoffee%3A%2F%2Flatte-macchiato`. The search
+   string URL-decoded then reads `?type=web+coffee://latte-macchiato`.
+
+## What this proposal is about
+
+The present _URL protocol handler registration for PWAs_ proposal is about offering protocol handler
+registration as part of a PWA installation through its manifest. After registering a PWA as a
+protocol handler, when a user clicks on a hyperlink with a specific scheme such as `mailto` ,
+`ms-word`, or `web+music` from a browser or a platform-specific app, the registered PWA will open
+and receive the URL. It is important to note that both the proposed manifest-based registration and
+the traditional `registerProtocolHandler()` play very similar roles in practice, while still
+allowing the possibility for complementary user-experiences:
+
+- Similarities include requirements around the list of schemes allowed to be registered, and
+  the name and format of parameters, etc.
+- Differences in the manifest-based registration are subtle, but might be useful to enhance the
+  experience for PWA users. For example, manifest-based PWA registration may not require an
```

Comment:
For security/privacy reasons we will always require additional user action, but it will be when invoking a link that is handled by an app. It will be in the form of a permission dialog (i.e.: allow or cancel) or a disambiguation dialog (i.e.: the open with dialog on Windows). Not sure if you want to talk about those cases in this document but it might be useful. If so we could help with screenshots and etc.

Reply:
[edit 1]: Ah, I see you talk about this further below.

Reply:
The Explainer is structured similarly.